gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/domain_service/CreateApiDomainService.java

```diff
@@ -15,16 +15,143 @@
+import io.gravitee.apim.core.notification.model.config.NotificationConfig;
+import io.gravitee.apim.core.parameters.model.ParameterContext;
+import io.gravitee.apim.core.parameters.query_service.ParametersQueryService;
+import io.gravitee.apim.core.search.Indexer;
+import io.gravitee.apim.core.workflow.crud_service.WorkflowCrudService;
+import io.gravitee.rest.api.model.parameters.Key;
+import io.gravitee.rest.api.model.parameters.ParameterReferenceType;
+import io.gravitee.rest.api.service.common.UuidString;
+import java.util.Collections;
 
-/**
- * @author Antoine CORDIER (antoine.cordier at graviteesource.com)
- * @author GraviteeSource Team
- */
-public interface CreateApiDomainService {
-    ApiWithFlows create(Api api, AuditInfo auditInfo);
-    Api create(ApiCRD api, AuditInfo auditInfo);
+public class CreateApiDomainService {
+
+    private final ValidateApiDomainService validateApiDomainService;
+    private final ApiCrudService apiCrudService;
+    private final AuditDomainService auditService;
+    private final ApiIndexerDomainService apiIndexerDomainService;
+    private final ApiPrimaryOwnerFactory apiPrimaryOwnerFactory;
+    private final ApiPrimaryOwnerDomainService apiPrimaryOwnerDomainService;
+    private final ApiMetadataDomainService apiMetadataDomainService;
+    private final FlowCrudService flowCrudService;
+    private final NotificationConfigCrudService notificationConfigCrudService;
+    private final ParametersQueryService parametersQueryService;
+    private final WorkflowCrudService workflowCrudService;
+
+    public CreateApiDomainService(
+        ValidateApiDomainService validateApiDomainService,
+        ApiCrudService apiCrudService,
+        AuditDomainService auditService,
+        ApiIndexerDomainService apiIndexerDomainService,
+        ApiMetadataDomainService apiMetadataDomainService,
+        ApiPrimaryOwnerFactory apiPrimaryOwnerFactory,
+        ApiPrimaryOwnerDomainService apiPrimaryOwnerDomainService,
+        FlowCrudService flowCrudService,
+        NotificationConfigCrudService notificationConfigCrudService,
+        ParametersQueryService parametersQueryService,
+        WorkflowCrudService workflowCrudService
+    ) {
+        this.validateApiDomainService = validateApiDomainService;
+        this.apiCrudService = apiCrudService;
+        this.auditService = auditService;
+        this.apiIndexerDomainService = apiIndexerDomainService;
+        this.apiPrimaryOwnerFactory = apiPrimaryOwnerFactory;
+        this.apiPrimaryOwnerDomainService = apiPrimaryOwnerDomainService;
+        this.apiMetadataDomainService = apiMetadataDomainService;
+        this.flowCrudService = flowCrudService;
+        this.notificationConfigCrudService = notificationConfigCrudService;
+        this.parametersQueryService = parametersQueryService;
+        this.workflowCrudService = workflowCrudService;
+    }
+
+    public ApiWithFlows create(Api api, AuditInfo auditInfo) {
```

Comment:
💥 this one is called by `CreateV4ApiUseCase` which already create and sanitize, meaning it will be done twice.

What about creating a new `SanitizedApi extends Api` that would be returned by the sanitizeService.
So, you can create a method overload `ApiWithFlows create(SanitizedApi api, AuditInfo auditInfo)` to skip the primaryOwner and sanitizing process
wdyt ?